pkg/tfbridge/provider.go

```diff
@@ -661,6 +661,15 @@ func (p *Provider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*pulum
 	doIgnoreChanges(ctx, res.TF.Schema(), res.Schema.Fields, olds, news, req.GetIgnoreChanges(), diff)
 	detailedDiff, changes := makeDetailedDiff(ctx, res.TF.Schema(), res.Schema.Fields, olds, news, diff)
 
+	// There are some providers/situations which `makeDetailedDiff` distorts the expected changes, leading
+	// to changes being dropped by Pulumi.
+	// Until we address https://github.com/pulumi/pulumi-terraform-bridge/issues/1501, it is safer to refer
+	// to the Terraform Diff attribute length for setting the DiffResponse.
+	// We will still use `detailedDiff` for diff display purposes.
+	if p.info.SkipDetailedDiffForChanges && len(diff.Attributes()) > 0 {
+		changes = pulumirpc.DiffResponse_DIFF_SOME
```

Comment:
Noting that it doesn't do the inverse, that is if len(diff.Attributes()) == 0 but makeDetailedDiff returned CHANGES, it does not override it back to NO_CHANGES. Is that perhaps a degenerate/ non-existent situation? Reading the code seems to be the case, so we're probably fine.

Reply:
We have not verified such a situation as of today to the best of my knowledge.